### ISSUE TITLE
[HOLD on #435128] WS chat is set as unread for Over Limit Approver when an expense/report is submitted to another approver

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#51723]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** N/A
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Issue reported by:** gknight@expensify.com
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1730281713256759

## Action Performed:

1. Create Control workspace, invite Submitter, Approver A, Approver B
2. Set approval workflow in OldDot as Advanced Approval + Submitter > Approver A + Approver B as $10 overLimitForwardsTo
3. As Submitter, create expense for + $10 in the WS chat

## Expected Result:
The WS chat should remain read for Approver B since the expense isn't ready for their attention.

## Actual Result:
The WS chat is unread for Approver B even though they have nothing to do.

## Workaround:
Nope!

## Platforms:
All

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MitchExpensify** [comment]
Replicated and issue seems unique 
<img width="802" alt="image" src="https://github.com/user-attachments/assets/8d498c32-5cbc-48ce-9343-d52128f59467">

**Comment by MitchExpensify** [comment]
Can this be external @Beamanator ?

**Comment by Beamanator** [comment]
Ehhhhh i think this should be internal, because externally we'd just do a bandaid fix most likely

**Comment by Beamanator** [comment]
I'm working on related things in https://github.com/Expensify/App/issues/51689 so i can probably take this today

**Comment by Beamanator** [comment]
FYI I'm seeing that this isn't specific to `overLimitForwardsTo`, I'm seeing even `forwardsTo` is seeing the WS chat unread when it should be, am i crazy?

**Comment by garrettmknight** [comment]
You might not be crazy! Jason reported for overLimit and I repro'd that. If this extends to `forwardsTo` let's fix there too.

**Comment by Beamanator** [comment]
Giggidy - I'll look into this now!

I would go either HIGH or MEDIUM priority for this one, what do y'all think?

**Comment by Beamanator** [comment]
Ok sorry but even one more point: I am pretty sure the `overLimitForwardsTo` got the Unread workspace when the expense was created / added to the workspace chat - even before the submitter submitted it!

**Comment by garrettmknight** [comment]
I'd say medium, it's confusing, not debilitating.

**Comment by garrettmknight** [comment]
Ah, you're on delay submission - so you're saying the Draft expense also set as unread? I don't think that should happen for anyone. Wanna clean that up now too?

**Comment by Beamanator** [comment]
Ok so a bit more clarity:
- Yes on delayed submission 👍 
- I thought the problem was about SUBMITting the report, but it's actually about SUBMITting the expense (which is a bit confusing since that's not actually submitting the report, it's just creating an expense on a new report, but it's not submitted yet)

I think i was previously confused by this issue title 😅  so submitting **THE EXPENSE** is what causes the unread 👍 , not submitting the report to any approver 👍

**Comment by garrettmknight** [comment]
> Also note: It even looks like the submitsTo sees the workspace chat as unread when the submitter just creates an expense on a draft report, BEFORE submitting it

This also seems wrong since they have nothing to do, but I can flag this more generally. I can confirm we've discussed not making the WS chat unread for further approvers when they don't have anything to review.

**Comment by garrettmknight** [comment]
@MitchExpensify once #435128 is done, let's confirm this works.

**Comment by Beamanator** [comment]
Aah shucks, ok then I'll 🏃  for now 👍 

Does this also wipe out the "and unread" part of the requirement of [this issue](https://github.com/Expensify/App/issues/51689)?

> # Expected Result:
> 
> Approver B should see that the WS Chat is GBR **and unread.**

**Comment by garrettmknight** [comment]
Yeah, I think it should, but we still need GBR to show.

**Comment by Beamanator** [comment]
Yep yep 👍  i just commented in the other issue, GBR fix should hopefully be live

**Comment by MitchExpensify** [comment]
Not overdue, waiting on https://github.com/Expensify/Expensify/issues/435128 to hit prod to test

**Comment by MitchExpensify** [comment]
Not overdue, waiting on https://github.com/Expensify/Expensify/issues/435128  to test

**Comment by MitchExpensify** [comment]
Testing now that https://github.com/Expensify/Expensify/issues/435128 is done:

1. Create Control workspace, invite Submitter, Approver A, Approver B
2. Set approval workflow in OldDot as Advanced Approval + Submitter > Approver A + Approver B as $10 overLimitForwardsTo

<img width="1344" alt="Image" src="https://github.com/user-attachments/assets/f0b9c407-fe2a-4016-bba7-ab1060083918" />

3. As Submitter, create expense for + $10 in the WS chat

<img width="1394" alt="Image" src="https://github.com/user-attachments/assets/c24459c3-594b-4b0b-b9be-ab090cc49899" />

**Comment by MitchExpensify** [comment]
Looks like this has been resolved based on the above test, closing!
